stl/inc/format

```diff
@@ -3417,8 +3417,10 @@ _NODISCARD auto make_wformat_args(_Args&&... _Vals) {
     return _Format_arg_store<wformat_context, _Args...>{_Vals...};
 }
 
-_EXPORT_STD template <output_iterator<const char&> _OutputIt>
-_OutputIt vformat_to(_OutputIt _Out, const string_view _Fmt, const format_args _Args) {
+_EXPORT_STD template <output_iterator<const char&> _OutputIt,
+    class _Sv = string_view> // lazy instantiation, see GH-1926
+_OutputIt vformat_to(_OutputIt _Out, const type_identity_t<_Sv> _Fmt, const format_args _Args) {
+    static_assert(is_same_v<_Sv, string_view>, "misspecified template argument");

```

Comment:
I'm a bit nervous about messing with the interface of these function templates, I'd like to avoid it if we can. What do you think about adding a type-dependent alias for `char` (resp. `wchar_t`) and parameterizing `_Format_handler` on that alias so that `_Parse_format_string` and `_Format_handler` both become type-dependent? This looks like:
```diff
+ using _CharX = conditional_t<is_same_v<_OutputIt, _Fmt_it>, char, char>;
  if constexpr (is_same_v<_OutputIt, _Fmt_it>) {
-     _Format_handler<char> _Handler(_Out, _Fmt, _Args);
+     _Format_handler<_CharX> _Handler(_Out, _Fmt, _Args);
      _Parse_format_string(_Fmt, _Handler);
      return _Out;
  } else {
      _Fmt_iterator_buffer<_OutputIt, char> _Buf(_STD move(_Out));
-     _Format_handler<char> _Handler(_Fmt_it{_Buf}, _Fmt, _Args);
+     _Format_handler<_CharX> _Handler(_Fmt_it{_Buf}, _Fmt, _Args);
      _Parse_format_string(_Fmt, _Handler);
      return _Buf._Out();
  }
```
(I've implemented and tested this on https://github.com/CaseyCarter/STL/tree/lazy-vformat_to, feel free to merge to your branch if you like this approach.)

Reply:
I've merged your branch. Thank you!

I guess we can use `decltype((void) _Out, char{})` to improve compiler throughput.

Reply:
> I guess we can use `decltype((void) _Out, char{})` to improve compiler throughput.

Ooh, nice improvement.